### ISSUE TITLE
Runs with remote errors should fail with error, not remain in progress. 

### ISSUE THREAD

**Issue by mccabete** [PecanProject/pecan#1622]
This is a more general problem that issue #1617 is an example of.  Runs that die sometime during the ssh process cause the job to remain in progress, refreshing the error every few minuets. This causes issues for two reasons: 1) It leaves a job running, 2) Deleting that run erases the record of the error.

**Comment by tonygardella** [comment]
Making this an epic issue and putting #1617 under it. Seems like there could be a few things wrong that will be issues and organizing them under this issue would be good.

**Comment by ashiklom** [comment]
One thought -- if we use R's `system` or `system2` call to run the `job.sh`, it should be possible to capture the error in R, which should make it easy to stop the run / trigger the appropriate error catch.

**Comment by mccabete** [comment]
I'll look into that

**Comment by mccabete** [comment]
@ashiklom  What was it that would be neat to look into? Ancible?

**Comment by ashiklom** [comment]
Yup, [Ansible](http://docs.ansible.com/ansible/latest/intro.html). I mentioned it a while ago somewhere, maybe Gitter, but it provides a really nice way to remotely execute specific tasks. Using it might require a re-write of our entire remote infrastructure, but it might make it more robust and secure. In particular, I really like the way it handles password authentication for things like our BU geo servers via [Vaults](http://docs.ansible.com/ansible/latest/vault.html).

In a nutshell, remote execution using Ansible would entail having the user provide an Ansible "playbook" which contains a lot of the same stuff we already have in the `config.php`.

**Comment by robkooper** [comment]
How would this work for runs started from the web interface?

**Comment by ashiklom** [comment]
I'm not sure yet, I haven't thought it all through, and I'm not convinced it's the way to go. Just throwing it out as an idea.

I think it would work more-or-less the same way as it does now to the user, except that server credentials would be stored in an Ansible Vault (just a password-protected text file) instead of being entered by the user. The config.php could just contain a path to the Ansible playbook and vault files. Actually running stuff through Ansible is just a matter of running system commands locally, which is easy to do in PHP.

**Comment by robkooper** [comment]
I agree it has to be better. I would like there to be options such as:
- username/password
- private key
- existing tunnel

Eventually it would be nice if we can have a service account that can be used, and we have a private key somewhere and we have users that are allowed to use this service account on certain hosts.

**Comment by ashiklom** [comment]
Check out some Ansible prototypes at [this repo](https://github.com/ashiklom/ansible_pecan).

**Comment by mccabete** [comment]
A few examples of runs that should probably fail outright while on remote, but don't: 

Early in pecan workflow failure (ED and Jules): 
https://github.com/PecanProject/pecan/issues/1547

**Comment by ashiklom** [comment]
Mental note: Very likely related issue is #1288.

**Comment by infotroph** [comment]
Closing this on the assumption #1288 tracks our best guess at a root cause. If you're seeing a related problem with the current version of PEcAn, please open a new issue.
